packages/@eris-boiler/common/src/modules/loadable/file-loader.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  promises as fs,
+} from 'fs'
+import { join } from 'path'
+import {
+  identity,
+} from '@utils/identity'
+
+export type LoadedFile = Record<string, unknown> & {filePath: string}
+
+export async function load (path: string): Promise<LoadedFile | undefined>
+export async function load (path: Array<string>): Promise<Array<LoadedFile>>
+export async function load (
+  path: string | Array<string>
+): Promise<LoadedFile | Array<LoadedFile>>
+export async function load (
+  path: string | Array<string>,
+): Promise<LoadedFile | undefined | Array<LoadedFile>> {
+  const loader = async (str: string): Promise<LoadedFile | undefined> => {
+    const fd = await fs.stat(str)
+
+    const importName = fd.isDirectory()
+      ? str
+      : /^(?!.*\.(?:d|spec|test)\.[jt]s$).*(?=\.[jt]s$)/.exec(str)?.[0]
+
+    if (!importName) {
+      return undefined
```

Comment:
maybe just return nothing? unless you want to be explicit

Reply:
explicit returns is set in tsconfig yes